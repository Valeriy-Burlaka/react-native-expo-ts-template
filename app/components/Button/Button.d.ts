import React from 'react';

export type ButtonProps = {
  onPress: () => void;
  text: string;
}

export const Button: React.ComponentType<ButtonProps>;
