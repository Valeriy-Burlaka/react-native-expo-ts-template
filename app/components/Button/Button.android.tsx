import React from 'react';
import { Platform } from 'react-native';

import { type ButtonProps as Props } from './Button';

import { BaseButton } from './Button.default';

export const Button = ({ text, onPress }: Props) => {
  return (
    <BaseButton
      backgroundColor="orange"
      onPress={onPress}
      text={text + ` (${Platform.OS})`}
      style={{
        borderRadius: 4,
      }}
    />
  );
};
