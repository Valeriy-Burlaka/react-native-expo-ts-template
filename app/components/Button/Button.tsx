import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

interface Props {
  backgroundColor?: string;
  onPress: () => void;
  text?: string;
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 40,
    paddingVertical: 20,
  },
  text: {
    color: 'black',
  },
});

export const Button = ({ backgroundColor = 'violet', text = 'click me', onPress, }: Props) => {
  return (
    <TouchableOpacity
      style={{
        ...styles.container,
        backgroundColor,
      }}
      onPress={onPress}
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};
