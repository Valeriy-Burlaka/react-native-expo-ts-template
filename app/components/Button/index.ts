// import { Button } from "./Button";

export * from './Button';