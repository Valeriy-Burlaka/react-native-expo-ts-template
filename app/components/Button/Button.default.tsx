import React from 'react';
import { TouchableOpacity, Text, StyleSheet, ViewStyle } from 'react-native';

import { type ButtonProps as BaseProps } from './Button';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    maxHeight: 80,
    paddingHorizontal: 20,
    paddingVertical: 20,
    borderRadius: 6,
  },
  text: {
    color: 'black',
    fontFamily: 'space-mono',
    fontSize: 18,
    fontWeight: '500',
    letterSpacing: 1.25,
    textTransform: 'uppercase',
  },
});

export type Props = {
  backgroundColor: string;
  style?: ViewStyle;
}

export const BaseButton = ({ text, onPress, backgroundColor, style = {}}: BaseProps & Props) => {
  return (
    <TouchableOpacity
      style={{
        ...styles.container,
        ...style,
        backgroundColor,
      }}
      onPress={onPress}
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};
