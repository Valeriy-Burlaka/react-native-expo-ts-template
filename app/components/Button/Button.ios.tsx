import React from 'react';
import { Platform } from 'react-native';

import { type ButtonProps as Props } from './Button';

import { BaseButton } from './Button.default';

export const Button = ({ text, onPress }: Props) => {
  return (
    <BaseButton
      backgroundColor="slateblue"
      onPress={onPress}
      text={text + ` (${Platform.OS})`}
      style={{
        borderRadius: 16,
        shadowColor: 'dodgerblue',
        shadowOffset: { width: 1.5, height: 1.5 },
        shadowOpacity: 0.5,
        shadowRadius: 16,
        marginTop: 12,
      }}
    />
  );
};
