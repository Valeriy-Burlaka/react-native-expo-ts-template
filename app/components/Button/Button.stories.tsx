import React from "react";
import { View } from "react-native";
import { ComponentStory, ComponentMeta } from "@storybook/react-native";
import { Button } from "./Button";

const ButtonMeta: ComponentMeta<typeof Button> = {
  title: "Button",
  component: Button,
  argTypes: {
    onPress: { action: "pressed the button" },
  },
  decorators: [
    (Story) => (
      <View
        style={{
          borderColor: 'coral',
          borderStyle: 'solid',
          borderWidth: 1,
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center',
          maxWidth: '50vw',
          padding: '10%',
        }}
      >
        <Story />
      </View>
    ),
  ],
};

export default ButtonMeta;

type ButtonStory = ComponentStory<typeof Button>;

export const Default: ButtonStory = (args) => <Button {...args} />;
export const WithText = Default.bind({});
Default.args = {
  text: 'hello',
};

export const Pink = Default.bind({});
Pink.args = {
  text: 'hey',
  backgroundColor: 'pink',
};
