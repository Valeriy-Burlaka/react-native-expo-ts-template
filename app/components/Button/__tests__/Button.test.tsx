import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from 'components/Button';

test('renders correctly', () => {
  const tree = renderer.create(
    <Button onPress={() => console.log('Tested!')} />
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
});
