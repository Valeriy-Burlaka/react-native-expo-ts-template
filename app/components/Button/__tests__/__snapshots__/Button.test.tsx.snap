// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "violet",
      "opacity": 1,
      "paddingHorizontal": 40,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
      }
    }
  >
    click me
  </Text>
</View>
`;
